supportedBackends/google/pipelines/v2alpha1/src/main/scala/cromwell/backend/google/pipelines/v2alpha1/PipelinesApiAsyncBackendJobExecutionActor.scala

```diff
@@ -137,3 +245,16 @@ class PipelinesApiAsyncBackendJobExecutionActor(standardParams: StandardAsyncExe
     List(jesFileOutput)
   }
 }
+
+object PipelinesApiAsyncBackendJobExecutionActor {
+  private val gcsPathMatcher = "^gs://?([^/]+)/.*".r
+
+  private [v2alpha1] def groupParametersByBucket[T <: PipelinesParameter](parameters: List[T]): Map[String, List[T]] = {
+    parameters.foldRight(Map[String, List[T]]().withDefault(_ => List.empty)) { case (p, acc) =>
```

Comment:
if you 
```
import cats.syntax.foldable._
import cats.instances.map._
import cats.instances.list._
```
you can
```suggestion
    parameters.map(p =>
      p.cloudPath.toString match {
        case gcsPathMatcher(bucket) =>
          Map(bucket -> List(p))
      }
    ).combineAll
```

Reply:
Though that gets us further away from the request to use a `NonEmptyList` (which I could accommodate with a better choice of default), plus arguably tougher to read